javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -263,101 +272,114 @@ public Builder toBuilder() {
         return builder;
     }
 
-    void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifiers) throws IOException {
-        // Nested classes interrupt wrapped line indentation. Stash the current wrapping state and put
-        // it back afterwards when this type is complete.
-        int previousStatementLine = codeWriter.statementLine;
-        codeWriter.statementLine = -1;
-
-        try {
-            if (enumName != null) {
-                codeWriter.emitJavadoc(javadoc);
-                codeWriter.emitAnnotations(annotations, false);
-                codeWriter.emit("$L", enumName);
-                if (!anonymousTypeArguments.formatParts().isEmpty()) {
-                    codeWriter.emit("(");
-                    codeWriter.emit(anonymousTypeArguments);
-                    codeWriter.emit(")");
-                }
-                if (fieldSpecs.isEmpty() && methodSpecs.isEmpty() && typeSpecs.isEmpty()) {
-                    return; // Avoid unnecessary braces "{}".
-                }
-                codeWriter.emit(" {\n");
-            } else if (anonymousTypeArguments != null) {
-                TypeName supertype = !superinterfaces.isEmpty() ? superinterfaces.get(0) : superclass;
-                codeWriter.emit("new $T(", supertype);
+    /**
+     * Emits the header of the type spec and returns if the body should be emitted.
+     */
+    private boolean emitHeader(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifiers)
```

Comment:
I don't think there's any reason to pull this out into a separate method. At least, it's outside the scope of this PR. Can we revert this?